### ISSUE TITLE
Support beaker facts in the test matrix

### ISSUE THREAD

**Issue by ekohl** [voxpupuli/gha-puppet#6]
https://github.com/voxpupuli/gha-puppet/blob/6602b90edacc1ef5e054cc15da0be54eec252288/.github/workflows/beaker.yml#L80

It's possible to set a fact via environment variables. This is used in [theforeman/puppet-pulpcore](https://github.com/theforeman/puppet-pulpcore/blob/d347abdac8091ed725a532427c3aa380a63bcbc9/.github/workflows/ci.yml#L57-L86) to run the entire test suite on multiple supported versions.

It would be great if this was also supported in gha-puppet.

I started with https://github.com/voxpupuli/puppet_metadata/pull/35 which at least pushes the name generation into the gem. That gem already generates the matrix so this opens the path to generating a bigger matrix. Then it just needs a good way to specify the inputs and map it.

**Comment by ekohl** [comment]
https://github.com/voxpupuli/puppet_metadata/pull/35 is merged and released in puppet_metadata 3.3.0.

That leaves the question of how to specify it. I'm leaning to a way to specify it in puppet_metadata. Either on the command line or via some config file. The config file has the benefit that we don't need to do some funky stuff in GHA, where you would need to use JSON to pass data structures around. It has the downside of maintaining an additional file in the repository.

I haven't investigated if we can actually added data to `metadata.json`. My use cases are that puppet-pulpcore supports some versions of Pulpcore and the same for puppet-candlepin. Listing those supported versions in `metadata.json` feels correct, because we also list the Puppet version as well as supported OS versions.

**Comment by ekohl** [comment]
> [voxpupuli/puppet_metadata#35](https://github.com/voxpupuli/puppet_metadata/pull/35) is merged and released in puppet_metadata 3.3.0.

And https://github.com/voxpupuli/gha-puppet/pull/41 uses this functionality.

**Comment by ekohl** [comment]
And https://github.com/theforeman/forklift/pull/1712#issuecomment-1751656340 explains how puppet-pulpcore uses this today.

**Comment by evgeni** [comment]
https://www.puppet.com/docs/puppet/7/modules_metadata.html#modules_metadata-metadata-version-requirement says:
> The requirements key specifies external requirements for the module, particularly the Puppet version required. Although you can express any requirement here, the Forge module pages and search function support only the "puppet" value, which specifies the Puppet version.

Which makes me think that specifying 
```json
  "requirements": [
    {
      "name": "puppet",
      "version_requirement": ">= 7.0.0 < 8.0.0"
    },
    {
      "name": "pulpcore",
      "version_requirement": "3.21 || 3.22 || 3.28"
    }
  ]
```
should totally work.

(the same wording is also used on the Puppet 6 and Puppet 8 pages)

**Comment by ekohl** [comment]
Not sure why it isn't closed, but #42 was merged.
